components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/EventDistributionTaskTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.event;
+
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+import static org.mockito.Matchers.any;
+
+
+public class EventDistributionTaskTest extends PowerMockTestCase{
+
+    @Mock
+    AbstractEventHandler module;
+
+    @Test
+    public void testRun() throws IdentityEventException {
```

Comment:
Throw generic Exception. In tests you don't need to throw specific types. (This is just for tests :))